src/measure_theory/function/conditional_expectation.lean

```diff
@@ -696,57 +696,59 @@ variables [complete_space E] {m m0 : measurable_space α} {μ : measure α}
 local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
 local notation `⟪`x`, `y`⟫₂` := @inner 𝕜 (α →₂[μ] E) _ x y
 
-variables (𝕜)
+variables (E 𝕜 μ)
 /-- Conditional expectation of a function in L2 with respect to a sigma-algebra -/
 def condexp_L2 (hm : m ≤ m0) : (α →₂[μ] E) →L[𝕜] (Lp_meas E 𝕜 m 2 μ) :=
 @orthogonal_projection 𝕜 (α →₂[μ] E) _ _ (Lp_meas E 𝕜 m 2 μ)
   (by { haveI : fact (m ≤ m0) := ⟨hm⟩, exact infer_instance, })
-variables {𝕜}
+variables {E 𝕜 μ}
 
 lemma ae_measurable'_condexp_L2 (hm : m ≤ m0) (f : α →₂[μ] E) :
-  ae_measurable' m (condexp_L2 𝕜 hm f) μ :=
+  ae_measurable' m (condexp_L2 E 𝕜 μ hm f) μ :=
 Lp_meas.ae_measurable' _
 
 lemma integrable_on_condexp_L2_of_measure_ne_top (hm : m ≤ m0) (hμs : μ s ≠ ∞) (f : α →₂[μ] E) :
-  integrable_on (condexp_L2 𝕜 hm f) s μ :=
-integrable_on_Lp_of_measure_ne_top ((condexp_L2 𝕜 hm f) : α →₂[μ] E)
+  integrable_on (condexp_L2 E 𝕜 μ hm f) s μ :=
+integrable_on_Lp_of_measure_ne_top ((condexp_L2 E 𝕜 μ hm f) : α →₂[μ] E)
   fact_one_le_two_ennreal.elim hμs
 
 lemma integrable_condexp_L2_of_is_finite_measure (hm : m ≤ m0) [is_finite_measure μ]
   {f : α →₂[μ] E} :
-  integrable (condexp_L2 𝕜 hm f) μ :=
+  integrable (condexp_L2 E 𝕜 μ hm f) μ :=
 integrable_on_univ.mp $ integrable_on_condexp_L2_of_measure_ne_top hm (measure_ne_top _ _) f
 
-lemma norm_condexp_L2_le_one (hm : m ≤ m0) : ∥@condexp_L2 α E 𝕜 _ _ _ _ _ _ _ _ μ hm∥ ≤ 1 :=
+lemma norm_condexp_L2_le_one (hm : m ≤ m0) : ∥condexp_L2 E 𝕜 μ hm∥ ≤ 1 :=
 by { haveI : fact (m ≤ m0) := ⟨hm⟩, exact orthogonal_projection_norm_le _, }
 
-lemma norm_condexp_L2_le (hm : m ≤ m0) (f : α →₂[μ] E) : ∥condexp_L2 𝕜 hm f∥ ≤ ∥f∥ :=
-((@condexp_L2 _ E 𝕜 _ _ _ _ _ _ _ _ μ hm).le_op_norm f).trans
+lemma norm_condexp_L2_le (hm : m ≤ m0) (f : α →₂[μ] E) :
+  ∥condexp_L2 E 𝕜 μ hm f∥ ≤ ∥f∥ :=
+((condexp_L2 E 𝕜 μ hm).le_op_norm f).trans
   (mul_le_of_le_one_left (norm_nonneg _) (norm_condexp_L2_le_one hm))
```

Comment:
This was already on the cusp of elaborator issues before - but now since almost all the lemmas have the problem this ran into, it's easier to just make `E` and `μ` explicit.

Reply:
Also, since `condexp_L2` is only an intermediate construction that should see no use beyond this file, having more explicit arguments to type will bother nobody.

Reply:
I hadn't really noticed that, that's great news

Reply:
Turns out I didn't need this change anyway, the linter spotted the underlying issue for me.